test/JavaScriptTests/patchConsoleObjectTests.js

```diff
@@ -0,0 +1,99 @@
+var assert = require('assert');
+var patchConsoleObject = require("../../bin/Debugger.Sample/PatchConsoleObject.js")
+
+var consoleOutput = "";
+
+var consoleDebugger = {
+    log(msg){
+        consoleOutput += "DEBUGGER LOG: "+ msg + "\n";
+    },
+    error(msg){
+        consoleOutput += "DEBUGGER ERROR: "+ msg + "\n";
+    },
+    warn(msg){
+        consoleOutput += "DEBUGGER WARN: "+ msg + "\n";
+    },
+    debug(msg){
+        consoleOutput += "DEBUGGER DEBUG: "+ msg + "\n";
+    },
+    info(msg){
+        consoleOutput += "DEBUGGER INFO: "+ msg + "\n";
+    }
+};
+
+var consoleObjectEmpty = {
+   
+};
+
+var consoleLogOnly = {
+    log(msg){
+        consoleOutput += "CONSOLE LOG: "+ msg + "\n";
+    }
+};
+
+var consoleObjectWithNoLoggingMethods = {
+    noLog(msg){
+        consoleOutput += "CONSOLE NOLOG: "+ msg + "\n";
+    }
+};
+
+var consoleObjectCompelete = {
+    log(msg){
+        consoleOutput += "CONSOLE LOG: "+ msg + "\n";
+    },
+    error(msg){
+        consoleOutput += "CONSOLE ERROR: "+ msg + "\n";
+    },
+    warn(msg){
+        consoleOutput += "CONSOLE WARN: "+ msg + "\n";
+    },
+    debug(msg){
+        consoleOutput += "CONSOLE DEBUG: "+ msg + "\n";
+    },
+    info(msg){
+        consoleOutput += "CONSOLE INFO: "+ msg + "\n";
+    }
+};
+
+describe("patchConsoleObjectTests", function() {
+    var global = {};
+    beforeEach(function() {
+        global = {};
+        consoleOutput = "";
+    });
+    it("case with empty console object", function() {
+        patchConsoleObject(global, consoleObjectEmpty, consoleDebugger);
```

Comment:
I didn't understand the point of these tests. Are these to test the console objects or the patch function logic?

Reply:
These tests are for testing patch function logic.

Reply:
@akroshg I wanted to be able to have tests to ensure the patching function works in the various cases. Let us know what you think.

Reply:
@akroshg Can you please complete this PR. Let me know if there is anything to change.